Tests/DistributedActorsTests/Cluster/SWIM/SWIMActorShellClusteredTests.swift

```diff
@@ -13,442 +13,469 @@
+        let firstNode = await self.setUpFirst()
+        let secondNode = await self.setUpSecond()
+        let thirdNode = await self.setUpThird()
 
-    func test_swim_shouldNotMarkUnreachable_whenSuspectedByNotEnoughNodes_whenMinTimeoutReached() async throws {
-        let first = await self.setUpFirst()
-        let firstNode = first.cluster.uniqueNode
-        let second = await self.setUpSecond()
+        guard let first = firstNode._cluster?._swimShell else {
+            throw testKit(firstNode).fail("SWIM shell of [\(firstNode)] should not be nil")
+        }
+        guard let second = secondNode._cluster?._swimShell else {
+            throw testKit(secondNode).fail("SWIM shell of [\(secondNode)] should not be nil")
+        }
+        guard let third = thirdNode._cluster?._swimShell else {
+            throw testKit(thirdNode).fail("SWIM shell of [\(thirdNode)] should not be nil")
+        }
+
+        try await self.configureSWIM(for: first, members: [second, third])
+
+        let throughPeer = try SWIMActorShell.resolve(id: second.id._asRemote, using: firstNode)
+        let targetPeer = try SWIMActorShell.resolve(id: third.id._asRemote, using: firstNode)
+
+        // FIXME: use non-response test probes instead of faking responses
+        // Fake a failed ping and ping request
+        _ = await first.whenLocal { __secretlyKnownToBeLocal in // TODO(distributed): rename once https://github.com/apple/swift/pull/42098 is implemented
+            // `first` got .timeout when pinging `third`, so it sends ping request to `second` for `three`
+            _ = __secretlyKnownToBeLocal.handlePingResponse(
+                response: .timeout(target: targetPeer, pingRequestOrigin: nil, timeout: .milliseconds(100), sequenceNumber: 13),
+                pingRequestOrigin: nil,
+                pingRequestSequenceNumber: nil
+            )
+            // `first` got .timeout from `second` for ping request
+            __secretlyKnownToBeLocal.handlePingRequestResponse(
+                response: .timeout(target: targetPeer, pingRequestOrigin: first, timeout: .milliseconds(100), sequenceNumber: 5),
+                pinged: throughPeer
+            )
+        }
 
-        first.cluster.join(node: second.cluster.uniqueNode.node)
-        try assertAssociated(first, withExactly: second.cluster.uniqueNode)
-        try assertAssociated(second, withExactly: first.cluster.uniqueNode)
+        // eventually it will ping/pingRequest and as all the pings (supposedly) time out it should mark as suspect
+        try await self.awaitStatus(.suspect(incarnation: 0, suspectedBy: [first.node]), for: targetPeer, on: first, within: .seconds(1))
+    }
 
-        let probeOnSecond = self.testKit(second).makeTestProbe(expecting: SWIM.Message.self)
-        let remoteMemberRef = first._resolveKnownRemote(probeOnSecond.ref, onRemoteSystem: second)
+    func test_swim_shouldNotMarkUnreachable_whenSuspectedByNotEnoughNodes_whenMinTimeoutReached() async throws {
```

Comment:
This test needs "unresponsive test probe" to work. There are other tests in this file that would be better done with such test probe, but I manage to make them work with real systems.

Reply:
Right sorry I was working towards it but then uncovered the compiler blockers... Will it back shortly!